### DISCUSSION TITLE
JIT session error: Cannot allocate memory

### DISCUSSION THREAD

**Discussion by JuanFMontesinos** [jax-ml/jax#22444] [Q&A]
Hi,
I was using jax and training a neural network for time series event detection.
In essence, at some point I have a non-jited function that calls the following code:
```
try:
    slice_data = softmax_prob[cc.start : cc.end, i]
except:
    print(f"Error: {cc.start}, {cc.end}, {i}, {softmax_prob.shape}")
    breakpoint()
```
where the try-except was added for debugging purposed. When trying to slice that array, it raises the `JIT session error: Cannot allocate memory`. 
With the debugger, I could check that, for that particular example, the values seems overall right:
<img width="575" alt="image" src="https://github.com/user-attachments/assets/e7d5f1d5-a82d-4bee-97c6-8edfa13e4832">

If I try to slice (once in the debugger), I obtain this 
<img width="489" alt="image" src="https://github.com/user-attachments/assets/805b1b5c-b652-424e-b1d0-93d49f753e6d">

<img width="475" alt="image" src="https://github.com/user-attachments/assets/299394ea-6d63-4f42-9aad-b87a2e9585c9">

I was also carefully following the system stats, there was enough space in tmp, more than 100 Gb RAM free, and 6 Gb GPU RAM still free too. Also this happened in after training iteration 8000+ and after testing iteration 800, which means it's an exotic error. 

Do anybody have any clue what could be the error?

**Comment by Qazalbash** [comment]
I had a similar [issue](https://github.com/google/jax/discussions/21375). It was resolved by setting `XLA_PYTHON_CLIENT_ALLOCATOR=platform` and changing some hyperparameters.
